realsense_camera/src/base_nodelet.cpp

```diff
@@ -554,10 +556,16 @@ namespace realsense_camera
     if (stream_index == RS_STREAM_DEPTH)
     {
       // fill depth buffer
-      image_depth16_ = reinterpret_cast <const uint16_t * >(rs_get_frame_data(rs_device_, RS_STREAM_DEPTH, 0));
+      image_depth16_ = reinterpret_cast <const uint16_t * >(rs_get_frame_data(rs_device_, stream_index, 0));
```

Comment:
Please remove the whitespace from the cast.
`reinterpret_cast<const uint16_t *>`